modules/social_features/social_event/src/Routing/EnrollRequestsOverviewSubscriber.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Drupal\social_event\Routing;
+
+use Drupal\Core\Url;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Class RedirectSubscriber.
+ *
+ * @package Drupal\social_event\Routing
+ */
+class EnrollRequestsOverviewSubscriber implements EventSubscriberInterface {
+
+  /**
+   * Get the request events.
+   *
+   * @inheritDoc
+   */
+  public static function getSubscribedEvents() {
+    $events[KernelEvents::REQUEST][] = ['checkAccessToEnrollRequestsOverview'];
+    return $events;
+  }
+
+  /**
+   * Check if the user is allowed to view this overview.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\GetResponseEvent $event
+   *   The event.
+   */
+  public function checkAccessToEnrollRequestsOverview(GetResponseEvent $event) {
```

Comment:
@frankgraave for this one you can better use a custom `views access` plugin as you did earlier on for the Event Invite feature :)